src/SFA.DAS.Reservations.Data/Repository/AccountLegalEntityRepository.cs

```diff
@@ -64,5 +65,29 @@ public async Task Remove(AccountLegalEntity accountLegalEntity)
                 transaction.Commit();
             }
         }
+
+        public async Task UpdateAccountLegalEntitiesToLevy(AccountLegalEntity accountLegalEntity)
+        {
+            using (var transaction = _dataContext.Database.BeginTransaction())
+            {
+                var accountLegalEntitiesList = await _dataContext.AccountLegalEntities
+                    .Where(x => x.AccountId.Equals(accountLegalEntity.AccountId))
+                    .ToListAsync();
+
+                if (accountLegalEntitiesList != null &&
+                    accountLegalEntitiesList.Any())
+                {
+                    accountLegalEntitiesList.ForEach(entity => entity.IsLevy = true);
+                    _dataContext.AccountLegalEntities.UpdateRange(accountLegalEntitiesList);
+                    _dataContext.SaveChanges();
+                }
+                else
+                {
+                    throw new DbUpdateException($"Record not found AccountId:{accountLegalEntity.AccountId}", (Exception)null);
```

Comment:
Can we add a log message to this as well - Will keep the the exception although might need to revisit this at some point.